frameworks/preact/setup.d.ts

```diff
@@ -0,0 +1,14 @@
+import type { AnyComponent } from "preact";
+
+export declare function setupPreviews<Props>(
```

Comment:
Custom previews are deprecated (see https://previewjs.com/docs/features/custom-previews), so we don't need to add them for Preact.

We can remove the `setup` files as well as all usage of `__previewjs_variants`.